src/visitpy/common/Logging.C

```diff
@@ -1640,19 +1649,112 @@ static std::string log_ResetPickAttributesRPC(ViewerRPC *rpc)
 
 static std::string log_AddAnnotationObjectRPC(ViewerRPC *rpc)
 {
-    return MESSAGE_COMMENT("AddAnnotationObject", MSG_NOT_IMPLEMENTED);
+    int annotType = rpc->GetIntArg1();
+    // annotType may have a 'visible/not visible' bit.
+    // this next is taken from VisWinAnnotations::AddAnnotationObject
+    // in order to remove the visible bit and get at just the annotType
+    int const static CREATE_ANNOTATION_OBJECT_AS_NOT_VISIBLE = 0x00010000;
+    std::string visible("1");
```

Comment:
What does this section do? I am confused.

Reply:
> I am confused.

So was I when I discovered that the annotType wasn't matching the possible valid values!

Take a look at  this section of code, where the annotation type and a possible `createAsVisible` flag are being combined into the argument passed to AddAnnotationObject as `annotTypeAndFlags`.  The logic I used to extract just the annotationType information was copied from VisWinAnnotations as mentioned in the comment.

https://github.com/visit-dav/visit/blob/39228d081ddbbdef7894ebcd9c1b500e521900ba/src/visitpy/common/visitmodule.C#L15620-L15685

Reply:
Interesting. It all seems rather complicated. I understand what you're doing now.